src/components/Calculator.js

```diff
@@ -1,44 +1,57 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 export default class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  handleClick = (e) => {
+    e.preventDefault();
+    const buttonName = e.target.value;
+    const { total, next, operation } = calculate(this.state, buttonName);
+    this.setState({ total, next, operation });
   }
 
   render() {
+    const { total } = this.state;
     return (
       <form>
-        <input type="text" className="result" value="0" />
+        <input type="text" className="result" value={total || 0} />
```

Comment:
- [ ] It would be better if you can make all numbers appear on the screen right now only the results that I can see on the screen like will confuse the user so please try to fix that